HiP-DataStore/Controllers/QueryHelper.cs

```diff
@@ -51,13 +51,56 @@ public static class QueryHelper
                 : query.Where(x => x.Timestamp > timestamp.Value);
         }
 
-        public static IQueryable<T> FilterByUser<T>(this IQueryable<T> query, ContentStatus status, IIdentity user) where T : ContentBase
+        public static IQueryable<T> FilterByUser<T>(this IQueryable<T> query, ContentStatus status, IIdentity User) where T : ContentBase
         {
-            bool isAllowedGetAll = UserPermissions.IsAllowedToGetAll(user, status);
+            bool isAllowedGetAll = UserPermissions.IsAllowedToGetAll(User, status);
             return query.FilterIf(!isAllowedGetAll, x =>
-                        ((status == ContentStatus.All) && (x.Status == ContentStatus.Published)) || (x.UserId == user.GetUserIdentity()));
+                        ((status == ContentStatus.All) && (x.Status == ContentStatus.Published)) || (x.UserId == User.GetUserIdentity()));
         }
         /// <summary>
+        /// Returns all entries that are located within the radius, defined within the entry itself, around the given latitude and longitude.
+        /// If none or only one coordinate is given, all entries are returned.
+        /// </summary>
+        public static IQueryable<T> FilterByLocation<T>(this IQueryable<T> query, float? latitude, float? longitude) where T : ContentBase
+        {
+            List<int> excludedIds = new List<int>();
+
+            if (typeof(Exhibit) == typeof(T))
```

Comment:
Since the method is only applicable to exhibits (at the moment), consider making the method non-generic:
```CSharp
public static IQueryable<Exhibit> FilterByLocation(this IQueryable<Exhibit> query, ...)
```
Then, with some LINQ-magic, the code could be reduced to something like:
```CSharp
public static IQueryable<Exhibit> FilterByLocation(this IQueryable<Exhibit> query, float? latitude, float? longitude)
{
    if (latitude != null && longitude != null)
    {
        return query.ToList()
            .Where(o => GetDistanceFromLatLonInKm(o.Latitude, o.Longitude, latitude.Value, longitude.Value) <= o.AccessRadius)
            .AsQueryable();
    }

    return query;
}
```